addons/mail/static/src/models/message_list_view_item.js

```diff
@@ -2,13 +2,36 @@
 
 import { registerModel } from '@mail/model/model_core';
 import { attr, one } from '@mail/model/model_field';
+import { clear } from '@mail/model/model_field_command';
 
 /**
  * Models a relation between a message list view and a message view where
  * message is used as iterating field.
  */
 registerModel({
     name: 'MessageListViewItem',
+    recordMethods: {
+        /**
+         * @private
+         * @returns {FieldCommand}
+         */
+        _computeNotificationMessageView() {
+            if (this.message.message_type === 'notification' && this.message.originThread.model === 'mail.channel') {
+                return {};
+            }
+            return clear();
+        },
+        /**
+         * @private
+         * @returns {FieldCommand}
+         */
+        _computeMessageView() {
+            if (this.message.message_type !== 'notification' || this.message.originThread.model !== 'mail.channel') {
```

Comment:
can just check truthy `thread.channel` instead of `thread.model === 'mail.channel'`